lib/msf/core/data_store.rb

```diff
@@ -216,6 +219,36 @@ def from_file(path, name = 'global')
     end
   end
 
+  #
+  # Return a deep copy of this datastore.
+  #
+  def copy
+    clone = self.class.new
```

Comment:
`clone` is a common method name (shallow copy), I would avoid using it as a variable name.

Reply:
ok

Reply:
I have been bitten by something like this before. Go Ruby or OOP?

Reply:
Sure but, to be fair, this has been called 'clone' for many years - I just copied this function from below since it used to only exist inside of ModuleDataStore. You just can't see it in the condensed diff view.

Reply:
(granted, just because the code is old, doesn't mean it's correct :)